idseq_dag/engine/pipeline_step.py

```diff
@@ -96,6 +97,28 @@ def uploading_results(self):
             idseq_dag.util.s3.upload_folder_with_retries(f, self.s3_path(f))
         self.status = StepStatus.UPLOADED
 
+    def update_stage_status_json(self):
+        log.write(f"Updating {self.stage_name}_status.json with step {self.name}")
+
+        # Fetch file from s3, open, and load as a json
+        s3_stage_status_file = f"{self.output_dir_s3}/{self.stage_name}_status.json"
```

Comment:
I think pipeline step just receive the status filename.
Better yet would be if pipeline flow could handle this. Pipeline step either could return the final status or update some variables like status, error (eventually progress).
Would also maybe solve the concurrency issue, if the pipeline flow controls all access to the status file. Wdyt?

Reply:
I'm not too sure how the step would alert the pipeline flow that an update has occurred — I think that having a shared lock makes more sense.

Reply:
It would typically be through a callback passed on to pipeline step... but lets wait for another opinion. The lock works.